pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,628 @@
+		}
+	}
+
+	client := &http.Client{Transport: transport}
+	return client
+}
+
+// copyFileWithProgressBar serves as a wrapper to copy the file with a progress bar
+func copyFileWithProgressBar(output *os.File, resp *http.Response) error {
+	barTemplate := fmt.Sprintf(`{{ "Downloading file:" }} {{counters . }} {{ cycle . %s }} {{speed . }}`, progressBarCycle)
+
+	// start bar based on our template
+	bar := pb.ProgressBarTemplate(barTemplate).Start(0)
+	defer bar.Finish()
+	rd := bar.NewProxyReader(resp.Body)
+	bar.Start()
+
+	_, err := io.Copy(output, rd)
+	return err
+}
+
+// getOrCreateTokenSecret obtains a token secret to be used along with the virtualMachineExport
+func getOrCreateTokenSecret(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport) (*k8sv1.Secret, error) {
+	// Generate a random, 20 char string to be used as a token
+	token := rand.String(20)
+
+	secret := &k8sv1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getExportSecretName(),
+			Namespace: vmexport.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(vmexport, schema.GroupVersionKind{
+					Group:   exportv1.SchemeGroupVersion.Group,
+					Version: exportv1.SchemeGroupVersion.Version,
+					Kind:    "VirtualMachineExport",
+				}),
+			},
+		},
+		Type: k8sv1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			secretTokenKey: []byte(token),
+		},
+	}
+
+	_, err := client.CoreV1().Secrets(vmexport.Namespace).Create(context.Background(), secret, metav1.CreateOptions{})
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		return nil, err
+	}
+
+	return secret, nil
+}
+
+// getTokenFromSecret extracts the token from the secret specified on the virtualMachineExport
+func getTokenFromSecret(client kubecli.KubevirtClient, namespace string) (string, error) {
+	secretName := getExportSecretName()
+	secret, err := client.CoreV1().Secrets(namespace).Get(context.Background(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	token := secret.Data[secretTokenKey]
+	return string(token), nil
+}
+
+// handleCreateFlags ensures that only compatible flag combinations are used with 'create' or with 'download' with the --create flag
+func handleCreateFlags(args []string) error {
+	if vm == "" && snapshot == "" && pvc == "" {
+		return fmt.Errorf(ErrRequiredExportType)
+	}
+
+	if vm != "" {
+		exportSource = k8sv1.TypedLocalObjectReference{
+			APIGroup: &virtv1.SchemeGroupVersion.Group,
+			Kind:     "VirtualMachine",
+			Name:     vm,
+		}
+	}
+	if snapshot != "" {
+		exportSource = k8sv1.TypedLocalObjectReference{
+			APIGroup: &snapshotv1.SchemeGroupVersion.Group,
+			Kind:     "VirtualMachineSnapshot",
+			Name:     snapshot,
+		}
+	}
+	if pvc != "" {
+		exportSource = k8sv1.TypedLocalObjectReference{
+			APIGroup: &k8sv1.SchemeGroupVersion.Group,
+			Kind:     "PersistentVolumeClaim",
+			Name:     pvc,
+		}
+	}
+
+	// These flags should only be checked when using 'create'
+	if funcName == CREATE {
```

Comment:
didn't we check that `funcName == CREATE` by virtue of calling handleCreateFlag. So this check seems unneeded.

Reply:
The idea is that this function is also used when using `download` with the `--create` flag, so I used that condition to avoid incompatibilities between the two. However I'll probably remove that flag, so the function will likely change.